src/qibocal/auto/history.py

```diff
@@ -34,16 +33,22 @@ class Completed:
     _results: Optional[Results] = None
     """Fitting output."""
 
-    @cached_property
+    @property
     def datapath(self):
         """Path contaning data and results file for task."""
         path = self.folder / "data" / f"{self.task.id}_{self.task.iteration}"
-        os.makedirs(path)
+        if not os.path.isdir(path):
+            os.makedirs(path)
         return path
 
     @property
     def results(self):
         """Access task's results."""
+        if not (self.datapath / RESULTSFILE).is_file():
+            return self._results
```

Comment:
I would return `self._results` every time that is not empty
```suggestion
        if self._results is not None:
            return self._results
```
No need of reloading.

If your goal is to also return `None` if there is no file, add it explicitly:

```suggestion
        if self._results is not None:
            return self._results
        if not (self.datapath / RESULTSFILE).is_file():
            return None
```